app/models/time_profile.rb

```diff
@@ -71,27 +71,29 @@ def tz_or_default(default_tz = DEFAULT_TZ)
   end
 
   def days
-    profile[:days]
+    profile["days"] || profile[:days]
   end
 
   def days=(arr)
     arr = arr.collect(&:to_i)
     profile_will_change! if profile[:days] != arr
     profile[:days] = arr
+    profile["days"] = arr
   end
 
   def hours
-    profile[:hours]
+    profile["hours"] || profile[:hours]
   end
 
   def hours=(arr)
     arr = arr.collect(&:to_i)
     profile_will_change! if profile[:hours] != arr
     profile[:hours] = arr
+    profile["hours"] = arr
   end
 
   def entire_tz?
-    days.sort == ALL_DAYS && hours.sort == ALL_HOURS
+    days.sort == ALL_DAYS && hours.sort == ALL_HOURS if !days.nil? && !hours.nil?
```

Comment:
Use Ruby's [Safe Navigation Operator](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/)?
```suggestion
    days&.sort == ALL_DAYS && hours&.sort == ALL_HOURS
```